Spatial.gd

```diff
@@ -690,30 +700,35 @@ func gen_adjustment_nodes():
 				continue
 			var new_gizmo = gizmo_scene.instance()
 			new_gizmo.translation = gizmo_position
-			new_gizmo.link_point("path", route, path2d, i)
-			new_gizmo.connect("selected", self, "on_gizmo_selected")
+			new_gizmo.connect("selected", self, "on_gizmo_selected", ["path", [route, path2d], i])
```

Comment:
I think something is too clever here passing in an array of [route, path2d] or [icon] depending on the first argument being "path" or "icon".